src/main/java/com/puppycrawl/tools/checkstyle/checks/imports/PkgImportControl.java

```diff
@@ -57,82 +51,58 @@
     private final Pattern patternForExactMatch;
     /** If this package represents a regular expression. */
     private final boolean regex;
-    /** Strategy in a case if matching allow/disallow rule was not found. */
-    private final MismatchStrategy strategyOnMismatch;
 
     /**
      * Construct a root node.
-     * @param pkgName the name of the package.
-     * @param regex flags interpretation of pkgName as regex pattern.
+     * @param name the name of the node.
+     * @param regex flags interpretation of name as regex pattern.
      * @param strategyOnMismatch strategy in a case if matching allow/disallow rule was not found.
      */
-    ImportControl(String pkgName, boolean regex,
-                  MismatchStrategy strategyOnMismatch) {
-        parent = null;
+    PkgImportControl(String name, boolean regex, MismatchStrategy strategyOnMismatch) {
+        super(null, strategyOnMismatch);
+
         this.regex = regex;
-        this.strategyOnMismatch = strategyOnMismatch;
         if (regex) {
-            // ensure that fullPackage is a self-contained regular expression
-            fullPackage = encloseInGroup(pkgName);
-            patternForPartialMatch = createPatternForPartialMatch(fullPackage);
-            patternForExactMatch = createPatternForExactMatch(fullPackage);
+            // ensure that fullName is a self-contained regular expression
+            fullName = encloseInGroup(name);
```

Comment:
Class and Package both have a name field. My original intention was to make `AbstractImportControl` house the `name` field, but because of the complex computations done to create the names for normal and regexp, they must remain in their respected classes. Leaving both as `name` instead of package/class is a leftover result of that.

I will fix up the naming some.

Reply:
Done.